src/flamegraph/mod.rs

```diff
@@ -325,19 +333,51 @@ impl Rectangle {
 ///
 /// [differential flame graph]: http://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html
 #[allow(clippy::cyclomatic_complexity)]
-pub fn from_sorted_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
+pub fn from_lines<'a, I, W>(opt: &mut Options, lines: I, writer: W) -> quick_xml::Result<()>
 where
     I: IntoIterator<Item = &'a str>,
     W: Write,
 {
-    let (bgcolor1, bgcolor2) = color::bgcolor_for(opt.bgcolors, opt.colors);
+    let mut reversed = StrStack::new();
+    let (mut frames, time, ignored, delta_max) = if !opt.no_sort && !opt.reverse_stack_order {
+        // Sort lines by default.
+        let mut lines: Vec<&str> = lines.into_iter().collect();
+        lines.sort_unstable();
+        merge::frames(lines)?
+    } else if opt.reverse_stack_order {
+        if opt.no_sort {
+            warn!("Input lines are always sorted when using --reverse. The --no-sort flag is being ignored.");
```

Comment:
This should already be enforced by the CLI opt parser, so this error will only appear for someone calling `from_lines` directly and setting `no_sort` and `reverse` in `Options`. We should probably warn about those fields instead of talking about command-line flags.